xbmc/pvr/dialogs/GUIDialogPVRTimerSettings.cpp

```diff
@@ -34,467 +39,1093 @@
+      if (channelEntry.second.clientId == pThis->m_timerType->GetClientId())
+        list.push_back(std::make_pair(channelEntry.second.description, channelEntry.first));
+    }
+
+    if (!foundCurrent && (pThis->m_channel == channelEntry.second))
+    {
+      current = channelEntry.first;
+      foundCurrent = true;
     }
   }
+}
 
-  CSettingInt *setting = AddSpinner(group, SETTING_TMR_CHNAME, 19078, 0, m_selectedChannelEntry, ChannelNamesOptionsFiller);
-  if (setting == NULL)
-    return NULL;
+// static
+void CGUIDialogPVRTimerSettings::DaysFiller(
+  const CSetting *setting, std::vector< std::pair<std::string, int> > &list, int &current, void *data)
+{
+  CGUIDialogPVRTimerSettings *pThis = static_cast<CGUIDialogPVRTimerSettings*>(data);
+  if (pThis == NULL)
+  {
+    CLog::Log(LOGERROR, "CGUIDialogPVRTimerSettings::DaysFiller - No dialog");
+    return;
+  }
 
-  return setting;
+  list.clear();
+  current = 0;
+
+  // Data range: "today" until "yesterday next year"
+  const CDateTime now(CDateTime::GetCurrentDateTime());
+  CDateTime time(now.GetYear(), now.GetMonth(), now.GetDay(), 0, 0, 0);
+  const CDateTime yesterdayPlusOneYear(
+    time.GetYear() + 1, time.GetMonth(), time.GetDay() - 1, time.GetHour(), time.GetMinute(), time.GetSecond());
+
+  time_t t(0);
+  while (time <= yesterdayPlusOneYear)
+  {
+    time.GetAsTime(t);
+    list.push_back(std::make_pair(time.GetAsLocalizedDate(), static_cast< int >(t)));
+    time += CDateTimeSpan(1, 0, 0, 0);
+  }
+
+  if (setting->GetId() == SETTING_TMR_FIRST_DAY)
+    current = GetDateAsInt(pThis->m_firstDayLocalTime);
+  else if (setting->GetId() == SETTING_TMR_START_DAY)
+    current = GetDateAsInt(pThis->m_startLocalTime);
+  else
+    current = GetDateAsInt(pThis->m_endLocalTime);
 }
 
-void CGUIDialogPVRTimerSettings::SetWeekdaySettingFromTimer(const CPVRTimerInfoTag &timer)
+// static
+void CGUIDialogPVRTimerSettings::DupEpisodesFiller(
+  const CSetting *setting, std::vector< std::pair<std::string, int> > &list, int &current, void *data)
 {
-  if (timer.m_bIsRepeating)
-  {
-    if (timer.m_iWeekdays == 0x01)
-      m_tmp_day = 0;
-    else if (timer.m_iWeekdays == 0x02)
-      m_tmp_day = 1;
-    else if (timer.m_iWeekdays == 0x04)
-      m_tmp_day = 2;
-    else if (timer.m_iWeekdays == 0x08)
-      m_tmp_day = 3;
-    else if (timer.m_iWeekdays == 0x10)
-      m_tmp_day = 4;
-    else if (timer.m_iWeekdays == 0x20)
-      m_tmp_day = 5;
-    else if (timer.m_iWeekdays == 0x40)
-      m_tmp_day = 6;
-    else if (timer.m_iWeekdays == 0x1F)
-      m_tmp_day = 7;
-    else if (timer.m_iWeekdays == 0x3F)
-      m_tmp_day = 8;
-    else if (timer.m_iWeekdays == 0x7F)
-      m_tmp_day = 9;
-    else if (timer.m_iWeekdays == 0x60)
-      m_tmp_day = 10;
+  CGUIDialogPVRTimerSettings *pThis = static_cast<CGUIDialogPVRTimerSettings*>(data);
+  if (pThis == NULL)
+  {
+    CLog::Log(LOGERROR, "CGUIDialogPVRTimerSettings::DupEpisodesFiller - No dialog");
+    return;
```

Comment:
Sigh, this is a pattern of the timer settings dialog original code. Thought, I would be on the safe side if I just do it like it was done before. Boy, this is making me sick. Feel free to change it and all the other nitpicking stuff, if you don't like it. Sorry, also having limited time myself.